### PR TITLE
Use mssql/server:2022-latest for MsSql container

### PR DESCRIPTION
**Pull request by cbeauchesne**
## Motivation

The MsSql image `mcr.microsoft.com/azure-sql-edge:latest` is very flaky at startuo, probably because of this issue : microsoft/mssql-docker#868

This image may not support kernel 6.7 and above, and we're using 6.8 kernel on github action large runners.

## Changes

Use `mcr.microsoft.com/mssql/server:2022-latest` image

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

**Comment by juan-fernandez** [comment]
just FYI, this change seems to make our tests pass: https://github.com/DataDog/dd-trace-js/pull/4710 (inspired by https://github.com/microsoft/mssql-docker/issues/868#issuecomment-2080122626)

**Comment by cbeauchesne** [comment]
Yes, the ticket says that the 2019 fix has been ported to 2022. Let;'s try with 2022.

File: utils/_context/containers.py [hunk 1]
```diff
@@ -918,21 +918,22 @@ def __init__(self, host_log_folder) -> None:
...
 
 
-class SqlServerContainer(SqlDbTestedContainer):
+class MsSqlServerContainer(SqlDbTestedContainer):
     def __init__(self, host_log_folder) -> None:
         self.data_mssql = f"./{host_log_folder}/data-mssql"
         healthcheck = {}
         if not platform.processor().startswith("arm"):
             # [!NOTE] sqlcmd tool is not available inside the ARM64 version of SQL Edge containers.
             # see https://hub.docker.com/_/microsoft-azure-sql-edge
```
Comment (vandonr) [inline utils/_context/containers.py:927]: I think with the new image we should remove this ?

Comment (cbeauchesne) [reply utils/_context/containers.py:927]: I didn't had the time to check, so I kept it in case. If you're able to verify that it's ok, you can remove it

Comment (vandonr) [reply utils/_context/containers.py:927]: I created https://github.com/DataDog/system-tests/pull/3108 as a follow up

**Pull request merged**